### PR TITLE
Work with CRUD Post

### PR DESCRIPTION
**Pull request by RafaelKhammadeev**
Close next items:
- User should be able to create a draft Article with title and rich text content
- User should be able to publish draft article

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+          --health-retries 5
+
+    steps:
+      - name: Run quality checks
+        run: |
+          bin/quality
+
+      - name: Setup db
+        run: |
+          sudo apt-get -yqq install libpq-dev
```
Comment (RasimKhusaenov) [inline .github/workflows/main.yml:28]: Do you need it?

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -0,0 +1,32 @@
+name: CI/CD
+on: push
+
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    services:
+      db:
+        image: postgres:11
```
Comment (RasimKhusaenov) [inline .github/workflows/main.yml:10]: Why 11?

File: app/controllers/posts_controller.rb [hunk 3]
```diff
@@ -1,56 +1,54 @@
 class PostsController < ApplicationController
+  before_action :set_user
   before_action :set_post, only: %i[show edit update destroy]
 
-  # GET /posts or /posts.json
   def index
-    @posts = Post.all
+    @posts = @user.posts
```
Comment (RasimKhusaenov) [inline app/controllers/posts_controller.rb:6]: Guest user can't see posts now, yeah?

File: app/views/layouts/application.html.erb [hunk 4]
```diff
@@ -6,6 +6,7 @@
     <%= csrf_meta_tags %>
     <%= csp_meta_tag %>
 
+    <link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/bootstrap@5.2.3/dist/css/bootstrap.min.css">
```
Comment (RasimKhusaenov) [inline app/views/layouts/application.html.erb:9]: Redundant?

File: app/views/posts/_form.html.erb [hunk 5]
```diff
@@ -1,4 +1,4 @@
-<%= form_with(model: post) do |form| %>
+<%= form_with(model: post, url:[user, post]) do |form| %>
```
Comment (RasimKhusaenov) [inline app/views/posts/_form.html.erb:1]:
```suggestion
<%= form_with [user, post] do |form| %>
```

File: app/views/posts/show.html.erb [hunk 6]
```diff
@@ -1,10 +1,20 @@
 <p style="color: green"><%= notice %></p>
 
-<%= render @post %>
+<div class="card" style="width: 18rem;">
```
Comment (RasimKhusaenov) [inline app/views/posts/show.html.erb:3]: It is bad practice to use tag `style`

File: app/views/users/index.html.erb [hunk 7]
```diff
@@ -2,13 +2,17 @@
...
 
-<div id="users">
+<table class="table">
+  <tbody>
   <% @users.each do |user| %>
-    <%= render user %>
-    <p>
-      <%= link_to "Show this user", user %>
-    </p>
+    <% next if user == current_user %>
```
Comment (RasimKhusaenov) [inline app/views/users/index.html.erb:8]: You should do it in controller

File: db/migrate/20230314211757_create_posts.rb [hunk 8]
```diff
@@ -1,7 +1,7 @@
 class CreatePosts < ActiveRecord::Migration[7.0]
   def change
     create_table :posts do |t|
-      t.string :text, null: false
+      t.string :description
```
Comment (RasimKhusaenov) [inline db/migrate/20230314211757_create_posts.rb:4]: You shouldn't change existing migrations

**Review by RasimKhusaenov — CHANGES_REQUESTED** [review]

File: .github/workflows/main.yml (hunk 1 above)
Comment (RafaelKhammadeev) [reply .github/workflows/main.yml:28]: I think, no

File: app/views/posts/index.html.erb [hunk 9]
```diff
@@ -1,14 +1,37 @@
...
+    <% else %>
+      <% @posts.each do |post| %>
+        <% next if post.status == "Draft" %>
+        <tr>
+          <th scope="row"><%= post.id %></th>
+          <td><%= link_to post.title, user_post_path(@user, post) %></td>
+          <td><%= post.description %></td>
+        </tr>
+      <% end %>
+    <% end %>
```
[the remark this answers is not in the document]
Comment (RafaelKhammadeev) [reply app/views/posts/index.html.erb:34]: I think it was better, so I do it in next PR

File: app/views/users/index.html.erb (hunk 7 above)
Comment (RafaelKhammadeev) [reply app/views/users/index.html.erb:8]: hm, yes cool idea, thanks

File: app/views/users/index.html.erb [hunk 10]
```diff
@@ -1,14 +1,12 @@
...
-<div id="users">
+<table class="table">
+  <tbody>
   <% @users.each do |user| %>
-    <%= render user %>
-    <p>
-      <%= link_to "Show this user", user %>
-    </p>
+    <tr>
+      <td><%= link_to "#{user.name} #{user.surname}", user_path(user) %></td>
```
Comment (RasimKhusaenov) [inline app/views/users/index.html.erb:7]: How about use decorator for it?
```suggestion
      <td><%= link_to user.full_name, user_path(user) %></td>
```

File: app/controllers/users_controller.rb [hunk 11]
```diff
@@ -1,47 +1,18 @@
 class UsersController < ApplicationController
-  before_action :set_user, only: %i[show edit update destroy]
+  before_action :set_user, only: %i[show destroy]
 
   # GET /users or /users.json
   def index
-    @users = User.all
+    @users = User.where.not(id: current_user)
```
Comment (RasimKhusaenov) [inline app/controllers/users_controller.rb:6]:
```suggestion
    @users = User.excluding(current_user)
```

File: app/policies/users/post_policy.rb [hunk 12]
```diff
@@ -0,0 +1,28 @@
...
+    def show?
+      true
+    end
+
+    def new?
+      user.present?
+    end
+
+    def edit?
+      true if record&.user_id == user.id
```
Comment (RasimKhusaenov) [inline app/policies/users/post_policy.rb:18]: Does it work?

File: config/routes.rb [hunk 13]
```diff
@@ -1,13 +1,12 @@
...
 
-  resources :posts
-  resources :users
-  # Define your application routes per the DSL in https://guides.rubyonrails.org/routing.html
+  get '/posts', to: 'posts#index'
 
-  # Defines the root path route ("/")
-  # root "articles#index"
+  resources :users, only: %i[index show destroy] do
+    namespace :users do
```
Comment (RasimKhusaenov) [inline config/routes.rb:8]: It seems incorrect

**Review by RasimKhusaenov — CHANGES_REQUESTED** [review]

**Pull request closed without merging**